### PR TITLE
(fix) O3-2025: Visit enabled system setting should be referenced using "visit.enabled"

### PR DESCRIPTION
**Pull request by Jexsie**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary

- Updates the Visit enabled system settings by name and if it's not set, defaults to true.

## Related Issue

- https://issues.openmrs.org/browse/O3-2025

### CODE REVIEW

**Comment by Jexsie** [comment]
cc @ibacher @mogoodrich

File: packages/esm-patient-medications-app/src/api/api.ts [hunk 1]
```diff
@@ -98,17 +98,14 @@ export function postOrder(body: OrderPost, abortController?: AbortController) {
...
-      ? `/ws/rest/v1/systemsetting/${config?.visitEnabledSystemSettingUUID}?v=custom:(value)`
-      : null,
+    `/ws/rest/v1/systemsetting/visit.enabled?v=custom:(value)`,
     openmrsFetch,
   );
 
   const results = useMemo(
     () => ({
-      systemVisitEnabled: data?.data?.value === 'true',
+      systemVisitEnabled: data?.data?.value ?? 'true',
```
Comment (ibacher) [inline packages/esm-patient-medications-app/src/api/api.ts:108]: I do think that defaulting this to `true` makes sense, but this (inadvertently) changes the type from a boolean to a string, which seems wrong:

```suggestion
      systemVisitEnabled: (data?.data?.value ?? 'true').toLowerCase() === 'true',
```

**Review by ibacher — APPROVED** [review]

**Comment by denniskigen** [comment]
This request is currently failing on dev3:

Screenshot 2023-04-18 at 11 59 11 PM

**Comment by denniskigen** [comment]
Potentially a similar situation to https://github.com/openmrs/openmrs-esm-patient-chart/pull/900

**Comment by mogoodrich** [comment]
@ibacher  @denniskigen  probably because that GP doesn't exist on dev3 (and that might be okay, as long as it just goes with the default, which I believe is 'true', if it isn't present?)

**Comment by Ruhanga** [comment]
@ibacher, @mogoodrich, @denniskigen, @Jexsie,  could we rather use the existing [visits.enabled](https://github.com/openmrs/openmrs-core/blob/b7c03a28c64493023a8211c53bb60d4d944b39b7/liquibase/src/test/resources/org/openmrs/liquibase/snapshots/core-data/liquibase-core-data-UPDATED-SNAPSHOT.xml#L3647) GP? Otherwise we update it to `visit.enabled` for consistency.

**Comment by vasharma05** [comment]
Shouldn't this change be done from the config schema itself, instead of writing it directly.

**Comment by vasharma05** [comment]
The UUID in the configuration should be replaced in the config schema to `visits.enabled`.

We shouldn't hardcode anything I believe, it should be done via configuration.

@Jexsie , please take note.

File: packages/esm-patient-medications-app/src/api/api.ts [hunk 2]
```diff
@@ -98,17 +98,14 @@ export function postOrder(body: OrderPost, abortController?: AbortController) {
 }
 
 export function useSystemVisitSetting() {
-  const config = useConfig() as ConfigObject;
   const { data, isLoading, error } = useSWRImmutable<FetchResponse<{ value: 'true' | 'false' }>, Error>(
-    config?.visitEnabledSystemSettingUUID
-      ? `/ws/rest/v1/systemsetting/${config?.visitEnabledSystemSettingUUID}?v=custom:(value)`
-      : null,
+    `/ws/rest/v1/systemsetting/visit.enabled?v=custom:(value)`,
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply packages/esm-patient-medications-app/src/api/api.ts:102]: CC: @mogoodrich @ibacher

**Comment by mogoodrich** [comment]
Yeah, I wrote the ticketed based on this line from the config description:

"_description: "UUID for the system setting's property 'visit.enabled'""... looks like there was typo there, if the global property is really called "visits.enabled", that's what we should be using.

@vasharma05 agree with @ibacher  Also in this case "visits.enabled" is really a configuration property defined on the back-end (as a global property, which is the way configuration properties are defined on the back end), so having a configuration property in the front-end config to point to "visits.enabled" is kind of duplicatively: having a configuration property that points to another configuration property.  Implementers have full control of what to set the 'visits.enabled" global property to.

**Pull request merged**